src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/VerificationComponents/VerifyGitopsEntities.tsx

```diff
@@ -0,0 +1,199 @@
+    identifier: entityIds?.cluster,
+    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier }
+  })
+  const {
+    loading: repoLoading,
+    refetch: refetchRepo,
+    error: repoFailed,
+    data: repoData
+  } = useAgentRepositoryServiceGet({
+    agentIdentifier: configdata?.agentInfo?.identifier as string,
+    identifier: entityIds?.repo,
+    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier }
+  })
+  const {
+    loading: applicationLoading,
+    refetch: refetchApp,
+    error: appFailed,
+    data: appData
+  } = useAgentApplicationServiceGet({
+    queryName: entityIds?.application,
+    agentIdentifier: configdata?.agentInfo?.identifier as string,
+    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier }
+  })
+
+  const setupState = React.useMemo((): PipelineSetupState => {
+    return stepsProgress?.[CDOnboardingSteps.DEPLOYMENT_STEPS]?.stepData
+  }, [stepsProgress])
+
+  useMemo((): string => {
+    const data = stepsProgress?.[CDOnboardingSteps.HOW_N_WHERE_TO_DEPLOY]?.stepData as WhereAndHowToDeployType
+    return data?.agentInfo?.identifier as string
+  }, [])
+
+  const successIconProps = React.useMemo((): { icon: IconName; iconProps: Omit<IconProps, 'name'> } => {
+    return {
+      icon: 'tick-circle',
+      iconProps: { padding: { right: 'medium' }, color: Color.GREEN_700 }
+    }
+  }, [])
+  const failIconProps = React.useMemo((): { icon: IconName; iconProps: Omit<IconProps, 'name'> } => {
+    return {
+      icon: 'danger-icon',
+      iconProps: { padding: { right: 'medium' }, color: Color.ERROR, className: css.redIcon }
+    }
+  }, [])
+
+  const loadingIconProps = React.useMemo((): { icon: IconName; iconProps: Omit<IconProps, 'name'> } => {
+    return {
+      icon: 'steps-spinner',
+      iconProps: { padding: { right: 'medium' } }
+    }
+  }, [])
+
+  useEffect(() => {
+    appData?.name &&
+      trackEvent(ONBOARDING_INTERACTIONS.APP_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
+    repoData?.repository &&
+      trackEvent(ONBOARDING_INTERACTIONS.REPO_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
+    clusterData?.cluster &&
+      trackEvent(ONBOARDING_INTERACTIONS.CLUSTER_VERIFICATION_SUCCESS, getBranchingProps(stepsProgress, getString))
+  }, [appData, clusterData, repoData])
+
+  useEffect(() => {
+    trackEvent(ONBOARDING_INTERACTIONS.APP_VERIFICATION_START, getBranchingProps(stepsProgress, getString))
+    trackEvent(ONBOARDING_INTERACTIONS.REPO_VERIFICATION_START, getBranchingProps(stepsProgress, getString))
+    trackEvent(ONBOARDING_INTERACTIONS.CLUSTER_VERIFICATION_START, getBranchingProps(stepsProgress, getString))
+  }, [])
+
+  useEffect(() => {
+    const allChecksPassed = [appFailed, repoFailed, clusterFailed].every(erorr => erorr === null)
+    const isEntityLoading = [applicationLoading, repoLoading, clusterLoading].includes(true)
```

Comment:
Might be easier to just use `||`
```suggestion
    const isEntityLoading = applicationLoading || repoLoading || clusterLoading
```